sql/src/main/java/org/apache/druid/sql/calcite/planner/SqlParameterizerShuttle.java

```diff
@@ -129,13 +129,12 @@ private SqlNode createArrayLiteral(Object value, int posn)
     List<SqlNode> args = new ArrayList<>(list.size());
     for (int i = 0, listSize = list.size(); i < listSize; i++) {
       Object element = list.get(i);
-      if (element == null) {
-        throw InvalidSqlInput.exception("parameter [%d] is an array, with an illegal null at index [%d]", posn + 1, i);
-      }
       SqlNode node;
-      if (element instanceof String) {
+      if (element == null) {
+        node = SqlLiteral.createNull(SqlParserPos.ZERO);
+      } else if (element instanceof String) {
         node = SqlLiteral.createCharString((String) element, SqlParserPos.ZERO);
-      } else if (element instanceof Integer || element instanceof Long) {
+      } else if (element instanceof Integer || element instanceof Long || element instanceof Double) {
```

Comment:
hmm, `createExactNumeric` doesn't really seem appropriate for `Double`, shouldn't that be handled with a separate else case, something like
```
...
      } else if (element instanceof Float || element instanceof Double) {
        node = SqlLiteral.createApproxNumeric(element.toString(), SqlParserPos.ZERO);
      }
...
```

Reply:
Even I thought so. But little worried about in filter converting them to Strings. Havent completely gone through it. Let me check and change it to approx numeric.